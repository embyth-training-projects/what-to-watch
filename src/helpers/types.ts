export interface ServerMovie {
  [`name`]: string;
  [`genre`]: string;
  [`released`]: number;
  [`background_color`]: string;
  [`background_image`]: string;
  [`preview_image`]: string;
  [`poster_image`]: string;
  [`id`]: number;
  [`description`]: string;
  [`rating`]: number;
  [`scores_count`]: number;
  [`director`]: string;
  [`starring`]: Array<string>;
  [`run_time`]: number;
  [`preview_video_link`]: string;
  [`video_link`]: string;
  [`is_favorite`]: boolean;
}

export interface ServerUser {
  [`id`]: number;
  [`email`]: string;
  [`name`]: string;
  [`avatar_url`]: string;
}

export interface MovieInterface {
  title: string;
  genre: string;
  date: number;
  background: string;
  backgroundColor: string;
  poster: string;
  id: number;
  description: string;
  rating: number;
  votes: number;
  director: string;
  starring: Array<string>;
  preview: string;
  previewImage: string;
  runTime: number;
  videoLink: string;
  isFavorite: boolean;
}

export interface ReviewInterface {
  comment: string;
  date: string;
  id: number;
  rating: number;
  user: {
    id: number;
    name: string;
  };
}

export interface UserInterface {
  id: number;
  name: string;
  email: string;
  avatarSrc: string;
}
