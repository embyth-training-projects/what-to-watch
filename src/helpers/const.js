export const NavTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const Pages = {
  MAIN: `main`,
  MOVIE: `movie`,
};

export const ALL_GENRES = `All genres`;
