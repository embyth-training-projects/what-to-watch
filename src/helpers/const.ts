export const ALL_GENRES = `All genres`;
export const VIDEO_PLAYER_ERROR_MESSAGE = `Sorry, your browser doesn't support embedded videos.`;
export const MOVIES_SHOWN = 8;
export const MOVIES_LIKE_THIS_SHOWN = 4;
export const MAX_SHOWN_GENRES = 10;

export enum NavTabs {
  OVERVIEW = `Overview`,
  DETAILS = `Details`,
  REVIEWS = `Reviews`,
}

export enum Pages {
  MAIN = `main`,
  MOVIE = `movie`,
  VIDEO = `video`,
  SIGN_IN = `sign-in`,
  ADD_REVIEW = `add-review`,
  MY_LIST = `my-list`,
  ERROR = `error`,
}

export enum CardVideoPlayerParams {
  WIDTH = `280`,
  HEIGHT = `175`,
}

export enum AppRoute {
  ROOT = `/`,
  SIGN_IN = `/login`,
  MY_LIST = `/mylist`,
  MOVIE = `/films`,
  ADD_REVIEW = `/review`,
  PLAYER = `/player`,
}

export enum Favorites {
  ADD = 1,
  REMOVE = 0,
}

export enum ErrorStatusCode {
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
  FORBIDDEN = 403,
  NOT_FOUND = 404,
  INTERNAL = 500,
  SERVICE_UNAVAILABLE = 503,
}

export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NOT_AUTH = `NOT_AUTH`,
}

export enum RequestStatus {
  NOT_SENT = `NOT_SENT`,
  SENDING = `SENDING`,
  SUCCESS = `SUCCESS`,
  ERROR = `ERROR`,
}

interface ApiConfigInterface {
  URL: string;
  TIMEOUT: number;
  COOKIES: boolean;
}

export const ApiConfig: ApiConfigInterface = {
  URL: `https://5.react.pages.academy/wtw`,
  TIMEOUT: 5000,
  COOKIES: true,
};

interface ReviewParamsInterface {
  MIN_LENGTH: number;
  MAX_LENGTH: number;
  RATINGS: Array<number>;
  BUTTON_LABEL: {
    POST: string;
    SENDING: string;
  };
}

export const Review: ReviewParamsInterface = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
  RATINGS: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
  BUTTON_LABEL: {
    POST: `Post`,
    SENDING: `Sending...`,
  },
};
