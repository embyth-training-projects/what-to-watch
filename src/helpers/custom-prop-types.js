import PropTypes from "prop-types";

const CustomPropTypes = {
  MOVIE: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    background: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    description: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    rating: PropTypes.string.isRequired,
    ratingDescription: PropTypes.string.isRequired,
    votes: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    preview: PropTypes.string.isRequired,
    runTime: PropTypes.string.isRequired,
  }).isRequired,

  REVIEW: PropTypes.shape({
    movie: PropTypes.string.isRequired,
    reviews: PropTypes.arrayOf(
        PropTypes.shape({
          author: PropTypes.string.isRequired,
          rating: PropTypes.string.isRequired,
          date: PropTypes.string.isRequired,
          content: PropTypes.string.isRequired,
          id: PropTypes.number.isRequired,
        }).isRequired
    ).isRequired,
  }).isRequired,
};

export default CustomPropTypes;
