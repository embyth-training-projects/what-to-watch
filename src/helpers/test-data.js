export const movieItemMock = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: 2014,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#eeeeee`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 190123,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: 8.9,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  previewImage: `img/bg-the-grand-budapest-hotel.jpg`,
  runTime: 99,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  isFavorite: false,
};

export const moviesMock = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Kids & Family`,
    date: 2018,
    background: `https://placeimg.com/1300/512/nature`,
    backgroundColor: `#eeeeee`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: 189234,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    votes: 500,
    director: `David Yates`,
    starring: [`Johnny Depp`, `Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewImage: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: 133,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    isFavorite: false,
  },
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: 2018,
    background: `https://placeimg.com/1300/512/nature`,
    backgroundColor: `#eeeeee`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
    rating: 8.0,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/f/fa/Tornado_time_lapse.webm`,
    previewImage: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: 134,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/f/fa/Tornado_time_lapse.webm`,
    isFavorite: false,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: 2004,
    background: `https://placeimg.com/1300/512/nature`,
    backgroundColor: `#eeeeee`,
    poster: `img/aviator.jpg`,
    id: 167456,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    rating: 7.5,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/9/9b/St._Lambertus_%28Immerather_Dom%29_2016.webm`,
    previewImage: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: 170,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/9/9b/St._Lambertus_%28Immerather_Dom%29_2016.webm`,
    isFavorite: false,
  },
  {
    title: `Shutter Island`,
    genre: `Thriller`,
    date: 2010,
    background: `https://placeimg.com/1300/512/nature`,
    backgroundColor: `#eeeeee`,
    poster: `img/shutter-island.jpg`,
    id: 156567,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    rating: 8.1,
    votes: 900,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/4/4d/Baha%27i_Temple_--_Wilmette_%2C_IL_--_Drone_Video_%28DJI_Spark%29.webm`,
    previewImage: `img/bg-the-grand-budapest-hotel.jpg`,
    runTime: 139,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/4/4d/Baha%27i_Temple_--_Wilmette_%2C_IL_--_Drone_Video_%28DJI_Spark%29.webm`,
    isFavorite: false,
  },
];

export const reviewItemMock = {
  user: {
    id: 0,
    name: `Kate Muir`,
  },
  rating: 8.9,
  date: `2019-05-08T14:13:56.569Z`,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  id: 345234523,
};

export const reviewsMock = [
  {
    user: {
      id: 0,
      name: `Kate Muir`,
    },
    rating: 8.9,
    date: `2019-05-08T14:13:56.569Z`,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    id: 345234523,
  },
  {
    user: {
      id: 1,
      name: `Max Miller`,
    },
    rating: 4.6,
    date: `2020-03-04T14:24:56.569Z`,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    id: 345255523,
  },
  {
    user: {
      id: 0,
      name: `John Feller`,
    },
    rating: 6.8,
    date: `2020-09-18T11:33:56.569Z`,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    id: 115234523,
  },
];

export const genres = [`All genres`, `Drama`, `Thriller`, `Comedy`];
