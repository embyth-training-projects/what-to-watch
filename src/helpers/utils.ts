import * as moment from "moment";

import {ReviewInterface} from "./types";

interface ExtendInterface {
  <T>(state: T, newStateValue: T): T;
}

export const extend: ExtendInterface = (state, newStateValue) => {
  return Object.assign({}, state, newStateValue);
};

export const getLeftColumnReviews = (reviews: Array<ReviewInterface>): Array<ReviewInterface> => {
  const sliceIndex = Math.ceil(reviews.length / 2);
  return reviews.slice(0, sliceIndex);
};

export const getRightColumnReviews = (reviews: Array<ReviewInterface>): Array<ReviewInterface> => {
  const sliceIndex = Math.ceil(reviews.length / 2);
  return reviews.slice(sliceIndex, reviews.length);
};

export const getRatingLevel = (rating: number): string => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  } else if (rating >= 3 && rating < 5) {
    return `Normal`;
  } else if (rating >= 5 && rating < 8) {
    return `Good`;
  } else if (rating >= 8 && rating < 10) {
    return `Very good`;
  } else if (rating === 10) {
    return `Awesome`;
  } else {
    return `Not rated`;
  }
};

export const getRunTimeFormat = (runTime: number): string => {
  const duration = moment.duration(runTime, `minutes`);
  return `${duration.get(`hours`)}h ${duration.get(`minutes`)}m`;
};

export const getReviewFormatDate = (date: string): string => {
  return moment(date).format(`MMMM D, YYYY`);
};

export const convertDateToISO = (date: string): string => {
  return moment(date).toISOString();
};

export const getTimeLeft = (totalTime: number, currentTime: number): string => {
  const duration = moment.duration(totalTime, `seconds`).subtract(moment.duration(currentTime, `seconds`));

  return moment.utc(duration.asMilliseconds()).format(`H:mm:ss`);
};

export const getFormatRating = (rating: number): string => {
  return (rating % 1 !== 0) ? rating.toString().split(`.`).join(`,`) : `${rating.toString()},0`;
};
