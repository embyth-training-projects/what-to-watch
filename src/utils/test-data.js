export const movieItemMock = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 194584,
  description: [`In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`, `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`],
  rating: `8,9`,
  ratingDescription: `Very good`,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  runTime: `1h 39m`,
};

export const moviesMock = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Kids & Family`,
    date: `2018`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: 189234,
    description: [`The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`],
    rating: `6,6`,
    ratingDescription: `Normal`,
    votes: 500,
    director: `David Yates`,
    starring: [`Johnny Depp`, `Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: `2h 13m`,
  },
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: `2018`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: [`The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`],
    rating: `8,0`,
    ratingDescription: `Good`,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/f/fa/Tornado_time_lapse.webm`,
    runTime: `2h 14m`,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: `2004`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    id: 167456,
    description: [`A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`],
    rating: `7,5`,
    ratingDescription: `Normal`,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/9/9b/St._Lambertus_%28Immerather_Dom%29_2016.webm`,
    runTime: `2h 50m`,
  },
  {
    title: `Shutter Island`,
    genre: `Thriller`,
    date: `2010`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/shutter-island.jpg`,
    id: 156567,
    description: [`In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`],
    rating: `8,1`,
    ratingDescription: `Good`,
    votes: 900,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/4/4d/Baha%27i_Temple_--_Wilmette_%2C_IL_--_Drone_Video_%28DJI_Spark%29.webm`,
    runTime: `2h 19m`,
  },
];

export const reviewsMock = [
  {
    movie: `The Grand Budapest Hotel`,
    reviews: [
      {
        author: `Kate Muir`,
        rating: `8,9`,
        date: `December 24, 2019`,
        content: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        id: 345234523,
      },
      {
        author: `Bill Goodykoontz`,
        rating: `8,0`,
        date: `November 18, 2015`,
        content: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
        id: 22345234,
      },
      {
        author: `Amanda Greever`,
        rating: `8,0`,
        date: `November 18, 2015`,
        content: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
        id: 4563456345,
      },
    ]
  },
  {
    movie: `Snatch`,
    reviews: [
      {
        author: `Kate Muir`,
        rating: `8,9`,
        date: `December 24, 2019`,
        content: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        id: 345234523,
      },
      {
        author: `Bill Goodykoontz`,
        rating: `8,0`,
        date: `November 18, 2015`,
        content: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
        id: 22345234,
      },
      {
        author: `Amanda Greever`,
        rating: `8,0`,
        date: `November 18, 2015`,
        content: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
        id: 4563456345,
      },
    ]
  },
];
