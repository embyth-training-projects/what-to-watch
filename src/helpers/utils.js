import moment from "moment";

import {ALL_GENRES} from "./const";

export const extend = (state, newStateValue) => {
  return Object.assign({}, state, newStateValue);
};

export const getMoviesGenres = (movies) => {
  return [ALL_GENRES, ...new Set(movies.map((movie) => movie.genre))];
};

export const getMovieReviews = (allReviews, currentMovie) => {
  return allReviews.filter((review) => review.movie === currentMovie.title)[0];
};

export const getMoreLikeThisMovies = (movies, currentMovie) => {
  return movies.filter((movie) => movie.genre === currentMovie.genre && movie.title !== currentMovie.title);
};

export const getLeftColumnReviews = (reviews) => {
  const sliceIndex = Math.ceil(reviews.length / 2);
  return reviews.slice(0, sliceIndex);
};

export const getRightColumnReviews = (reviews) => {
  const sliceIndex = Math.ceil(reviews.length / 2);
  return reviews.slice(sliceIndex, reviews.length);
};

export const filterMoviesByGenre = (allMovies, currentGenre) => {
  if (currentGenre === ALL_GENRES) {
    return allMovies;
  }

  return allMovies.filter((movie) => movie.genre === currentGenre);
};

export const getRatingLevel = (rating) => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  } else if (rating >= 3 && rating < 5) {
    return `Normal`;
  } else if (rating >= 5 && rating < 8) {
    return `Good`;
  } else if (rating >= 8 && rating < 10) {
    return `Very good`;
  } else if (rating === 10) {
    return `Awesome`;
  } else {
    return `Not rated`;
  }
};

export const getRunTimeFormat = (runTime) => {
  const duration = moment.duration(runTime, `minutes`);
  return `${duration.get(`hours`)}h ${duration.get(`minutes`)}m`;
};

export const getReviewFormatDate = (date) => {
  return moment(date).format(`MMMM D, YYYY`);
};

export const convertDateToISO = (date) => {
  return moment(date).toISOString();
};

export const getFormatRating = (rating) => {
  return (rating % 1 !== 0) ? rating.toString().split(`.`).join(`,`) : `${rating.toString()},0`;
};
