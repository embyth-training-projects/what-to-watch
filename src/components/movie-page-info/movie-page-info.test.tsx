import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import MoviePageInfo from "./movie-page-info";

import NameSpace from "../../store/name-space";
import {movieItemMock, reviewsMock, noop} from "../../helpers/test-data";

const mockStore = configureStore([]);

it(`MoviePageInfo should render correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      movieReviews: reviewsMock,
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <MoviePageInfo
            currentMovie={movieItemMock}
            currentActiveItem={`Reviews`}
            onItemClick={noop}
          />
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
