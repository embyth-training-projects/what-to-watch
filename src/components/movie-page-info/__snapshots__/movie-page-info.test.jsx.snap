// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageInfo should render correctly 1`] = `
<div
  className="movie-card__wrap movie-card__translate-top"
>
  <div
    className="movie-card__info"
  >
    <div
      className="movie-card__poster movie-card__poster--big"
    >
      <img
        alt="The Grand Budapest Hotel"
        height="327"
        src="img/the-grand-budapest-hotel-poster.jpg"
        width="218"
      />
    </div>
    <div
      className="movie-card__desc"
    >
      <nav
        className="movie-nav movie-card__nav"
      >
        <ul
          className="movie-nav__list"
        >
          <li
            className="movie-nav__item "
          >
            <a
              className="movie-nav__link"
              href="#"
              onClick={[Function]}
            >
              Overview
            </a>
          </li>
          <li
            className="movie-nav__item "
          >
            <a
              className="movie-nav__link"
              href="#"
              onClick={[Function]}
            >
              Details
            </a>
          </li>
          <li
            className="movie-nav__item movie-nav__item--active"
          >
            <a
              className="movie-nav__link"
              href="#"
              onClick={[Function]}
            >
              Reviews
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="movie-card__reviews movie-card__row"
      >
        <div
          className="movie-card__reviews-col"
        >
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Kate Muir
                </cite>
                <time
                  className="review__date"
                  dateTime="2019-12-23"
                >
                  December 24, 2019
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              8,9
            </div>
          </div>
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Bill Goodykoontz
                </cite>
                <time
                  className="review__date"
                  dateTime="2015-11-17"
                >
                  November 18, 2015
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              8,0
            </div>
          </div>
        </div>
        <div
          className="movie-card__reviews-col"
        >
          <div
            className="review"
          >
            <blockquote
              className="review__quote"
            >
              <p
                className="review__text"
              >
                I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
              </p>
              <footer
                className="review__details"
              >
                <cite
                  className="review__author"
                >
                  Amanda Greever
                </cite>
                <time
                  className="review__date"
                  dateTime="2015-11-17"
                >
                  November 18, 2015
                </time>
              </footer>
            </blockquote>
            <div
              className="review__rating"
            >
              8,0
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
